### PR TITLE
Add dialog that blocks the screen when connection is lost

### PR DESCRIPTION
**Pull request by beatrizz**
No internet behaviour.

There is a unique dialog for all the cases where we say there's no internet. It appears when the user presses the "create room" or "enter room" without internet or when connectivity is lost for more than 5 seconds during the game. If you press the "OK" button or the back button, you go back to the home page.  During the game, if the dialog appears and you recover connectivity while it's on screen, it disappears.

### CODE REVIEW

File: lib/widget/create_room_page.dart [hunk 1]
```diff
@@ -60,11 +61,16 @@ class CreateRoomPage extends StatelessWidget {
 
   Widget _createRoomButton(BuildContext context, Store<GameModel> store) => new RaisedButton(
         child: new Text(AppLocalizations.of(context).createRoom, style: buttonTextStyle),
-        onPressed: () {
-          FormState enterNameState = Keys.createRoomPageNameKey.currentState;
-          if (enterNameState.validate()) {
-            enterNameState.save();
-            store.dispatch(new CreateRoomAction());
+        onPressed: () async {
+          var connectivityResult = await (new Connectivity().checkConnectivity());
```
Comment (acjc) [inline lib/widget/create_room_page.dart]: If this check is slow, the user could probably see nothing happen when they tap the button, and try to tap it many times.

Maybe we should do the internet check inside the `CreateRoomAction` (which uses `withRequest()` to display a loading spinner).

File: lib/widget/home_page.dart [hunk 2]
```diff
@@ -40,13 +42,18 @@ class HomePage extends StatelessWidget {
...
         child: Text(AppLocalizations.of(context).enterRoom, style: buttonTextStyle),
-        onPressed: () {
-          FormState enterCodeState = Keys.homePageCodeKey.currentState;
-          FormState enterNameState = Keys.homePageNameKey.currentState;
-          if (enterCodeState.validate()) {
-            enterCodeState.save();
-            enterNameState.save();
-            store.dispatch(new ValidateRoomAction(context));
+        onPressed: () async {
+          var connectivityResult = await (new Connectivity().checkConnectivity());
```
Comment (acjc) [inline lib/widget/home_page.dart]: Same as above but inside `ValidateRoomAction`.

File: lib/widget/game.dart [hunk 3]
```diff
@@ -38,17 +42,41 @@ class Game extends StatefulWidget {
 
 class GameState extends State<Game> {
   final Store<GameModel> _store;
+  StreamSubscription _subscription;
```
Comment (acjc) [inline lib/widget/game.dart]: `connectivitySubscription` and `connectivityTimer`?

**Comment by beatrizz** [comment]
Resolves https://github.com/acjc/heist/issues/70

**Pull request merged**